packages/electron-chrome-extensions/src/browser/api/tabs.ts

```diff
@@ -143,6 +143,10 @@ export class TabsAPI {
   }
 
   private async create(event: ExtensionEvent, details: chrome.tabs.CreateProperties = {}) {
+    // make URL absolute
+    if (details.url) {
+      details.url = new URL(details.url, event.sender.getURL()).href;
```

Comment:
[According to chromium code](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/extensions/extension_tab_util.cc;l=745-752;drc=b3ca5c09fa0aa07b7f9921501f75e43d80f3ba48), it actually just resolves relative to the base extension URL.
```suggestion
      details.url = new URL(details.url, event.extension.url).href;
```

A couple other requests:
- Avoid mutating `details` if possible
- Can we do the same for `tabs.update`? The same logic is needed there
- Not required, but would be very appreciated if tests were added for relative URLs

Reply:
i very much prefer editing details since thats the object you use everywhere, but I've added a new object, updated the base url and also added it to tabs.update.